ui/src/resource-table.js

```diff
@@ -1,57 +1,190 @@
+      renderCell: params => {
+        return <ResultCell result={params.row.status} />;
+      }
     }
-  },
-  results: {
-    titles: ['Id', 'Build Id', 'Project Id', 'Result', 'Result Message', 'Test Name', 'Timestamp'],
-    keys: ['id', 'build_id', 'project_id', 'result', 'result_message', 'test_name', 'timestamp'],
-    links: {
-      project_id: (r) => `/projects/${r.project_id}/builds`,
-      test_name: (r) => `/projects/${r.project_id}/tests/history?name=${encodeURIComponent(r.test_name)}&count=10`
+  ],
+  results: [
+    {
+      field: 'id',
+      headerName: 'ID',
+      minWidth: 150,
+      sortable: false,
+      editable: false
+    },
+    {
+      field: 'build_id',
+      headerName: 'Build ID',
+      minWidth: 150,
+      sortable: false,
+      filterable: false,
+      renderCell: params => <Link href={`/projects/${params.row.id}/builds/${params.row.jobId}/result`}> {params.row.jobId} </Link>,
+      editable: false
+    },
+    {
+      field: 'result',
+      headerName: 'Result',
+      minWidth: 110,
+      renderCell: params => {
+        return <ResultCell result={params.row.result} />;
+      },
+      sortable: true
+    },
+    {
+      field: 'result_message',
+      headerName: 'Result Message',
+      sortable: false,
+      minWidth: 110,
+      flex: 1,
+      renderCell: params => (
+        <Link href={{ pathname: `/projects/${params.row.id}/builds/result`, query: { codeString: params.row.test_name } }}> {params.row.result_message} </Link>
+      )
+    },
+    {
+      field: 'test_name',
+      headerName: 'Test Name',
+      description: 'This column has a value getter and is not sortable.',
+      sortable: true,
+      minMinWidth: 260,
+      flex: 1,
+      renderCell: params => (
+        <Link href={`/projects/${params.row.project_id}/tests/history?name=${encodeURIComponent(params.row.test_name)}&count=10`}>
+          {' '}
+          {params.row.test_name}{' '}
+        </Link>
+      )
+    },
+    {
+      field: 'tags',
+      headerName: 'Tags',
+      sortable: true,
+      renderCell: params => {
+        if (params.row.tags && params.row.tags.nightly) {
+          return (
+            <div
+              style={{
+                color: 'orange'
+              }}
+            >
+              {' '}
+              nightly{' '}
+            </div>
+          );
+        }
+        return '';
+      },
+      minWidth: 100
+    },
+    {
+      field: 'timestamp',
+      headerName: 'TimeStamp',
+      sortable: true,
+      valueGetter: params => {
+        const date = new Date(params.row.timestamp * 1000);
+        return date.toLocaleDateString();
+      },
+      minWidth: 200
     }
-  }
+  ]
 };
 
 const ResourceTable = ({ resources, type }) => {
-  const { titles: columnTitles, keys: columnKeys, links: columnLinks } = tableTypes[type];
+  const columns = tableColumnDefinitions[type];
+
+  const rows = resources;
+
+  const router = useRouter();
+
+  var initialState = {};
```

Comment:
sorry that I didn't point this out earlier, but try to stay away from `var`s just as a general recommendation...